### DISCUSSION TITLE
How to filter objects with foreignkey field?

### DISCUSSION THREAD

**Discussion by Edw125** [piccolo-orm/piccolo#651] [General]
Hello everyone! I work with this framework recently and I have question: how to build a query that will filter objects Cohort by foreignkey field Cohort.course? In the photo, my request gives an error. Please tell me how to solve

![изображение](https://user-images.githubusercontent.com/57034089/197344528-593c74c6-24c1-48a3-a203-85fce097c97c.png)
![изображение](https://user-images.githubusercontent.com/57034089/197344669-81b07149-9147-4f00-b2da-f1e95a1db6ac.png)
![изображение](https://user-images.githubusercontent.com/57034089/197344683-0b8e314a-8158-4fd2-93e4-7d52f88dc981.png)

**Comment by dantownsend** [comment]
@Edw125 Hi - I think it's almost there. I can just spot one minor issue.

```python
cohort = await Cohort.objects().get_or_create(
   (Cohort.number == str(array[5])) & (Cohort.course.id == int(course.id))
)
```

With `get_and_create` you can only pass in a single value for the `where` clause at the moment (so you have to use `&`). We should probably change it.

Does that work for you? If not, can you provide what error is raised?

**Reply by Edw125** [reply]
Thank you very much, the suggested method helped

**Reply by dantownsend** [reply]
Awesome, glad it worked.
